### PR TITLE
[T4A4][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Email.java [hunk 1]
```diff
@@ -58,4 +58,9 @@ public int hashCode() {
     public boolean isPrivate() {
         return isPrivate;
     }
+
+	@Override
+	public String getPrintableString() {
+	    return "Email: " + value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Email.java:64]: you can consider eliminating the magic string here.

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -83,5 +83,23 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    /**
+     * Returns a string that contains the information of a Person.
+     */
+    public String getPrintableString() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java:90]: hmmm... this was unnecessary! think why?

File: src/seedu/addressbook/data/person/Person.java [hunk 3]
```diff
@@ -83,5 +83,23 @@ public int hashCode() {
...
+        return getPrintableString(name, phone, email, address); 
+    }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+    private String getPrintableString(Printable... printables) {
+    	String concatenateString  = printables[0].getPrintableString();
+    	for (int i = 1; i < printables.length; i++) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Person.java:99]: hehe! this is a hack :-P

File: src/seedu/addressbook/data/person/Printable.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package seedu.addressbook.data.person;
+
+/**
+ * A immutable interface to display a Peron's details
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Printable.java:4]: not necessary to be only person's details, no?
any class can implement this.
your header comment should be precise!

File: src/seedu/addressbook/data/person/Printable.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+ * Implementations should guarantee: details are present and not null, field values are validated.
+ */
+public interface Printable {
+	
+	String getPrintableString();
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Printable.java:9]: add a header comment to tell what this method is supposed to do?

**Comment by okkhoy** [comment]
@zyjames 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Person.java (hunk 3 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Person.java:99]: Yeap!(: A very good hack

File: src/seedu/addressbook/data/person/Printable.java (hunk 4 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Printable.java:4]: Yes, not only person's details. Will give a better comment for my class in future!(:

File: src/seedu/addressbook/data/person/Printable.java (hunk 5 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/Printable.java:9]: Yes, will add a header to tell what the method is doing.

**Pull request closed without merging**